projects/ngx-datatable/src/lib/utils/tree.ts

```diff
@@ -81,41 +85,31 @@ export function groupRowsByParents<TRow>(
     }
 
     let resolvedRows: any[] = [];
-    nodeById[0].flatten(function () {
-      resolvedRows = [...resolvedRows, this.row];
-    }, true);
+    nodeById[0].flatten(row => (resolvedRows = [...resolvedRows, row]));
 
     return resolvedRows;
   } else {
     return rows;
   }
 }
 
-class TreeNode {
-  public row: any;
-  public parent: any;
-  public children: any[];
+class TreeNode<TRow extends Row> {
+  public row: TRow;
+  public parent: TreeNode<TRow>;
+  public children: TreeNode<TRow>[];
 
-  constructor(row: any | null = null) {
-    if (!row) {
-      row = {
-        level: -1,
-        treeStatus: 'expanded'
-      };
-    }
+  constructor(row: TRow) {
     this.row = row;
     this.parent = null;
     this.children = [];
   }
 
-  flatten(f: any, recursive: boolean) {
+  flatten(f: (child: TreeNode<TRow>) => void) {
```

Comment:
Just a sidenote: We also drop the `recursive` parameter here, but the `flatten` method is only ever called in this file and the `recursive` parameter was never used.